omegaconf/omegaconf.py

```diff
@@ -829,6 +829,14 @@ def to_yaml(cfg: Any, *, resolve: bool = False, sort_keys: bool = False) -> str:
         :param sort_keys: If True, will print dict keys in sorted order. default False.
         :return: A string containing the yaml representation.
         """
+        if not (
+            is_primitive_container(cfg)
+            or is_structured_config(cfg)
+            or OmegaConf.is_config(cfg)
+        ):
+            raise ValueError(
+                "to_yaml expects input of type DictConfig / ListConfig / dict / list / StructuredConfig."
+            )
         cfg = _ensure_container(cfg)
         container = OmegaConf.to_container(cfg, resolve=resolve, enum_to_str=True)
         return yaml.dump(  # type: ignore
```

Comment:
The logic for the above if-statement is copied from the `_utils.py:_ensure_container` method, which looks like this:
```python
def _ensure_container(target: Any, flags: Optional[Dict[str, bool]] = None) -> Any:
    from omegaconf import OmegaConf

    if is_primitive_container(target):
        assert isinstance(target, (list, dict))
        target = OmegaConf.create(target, flags=flags)
    elif is_structured_config(target):
        target = OmegaConf.structured(target, flags=flags)
    assert OmegaConf.is_config(target)
    return target
```